R/zchunk_LA101.Energy_Balance_china.R

```diff
@@ -0,0 +1,172 @@
+      mutate(value = approx_fun(year, value, rule = 1)) %>%
+      ungroup() %>%
+      #if we have 1 data point, interpolate will replace it with NA, so we need to add those data back
+      left_join(L101.inNBS_Mtce_province_S_F %>% rename(org = value),
+                by = c("province", "fuel", "sector", "year")) %>%
+      mutate(value = replace(value, is.na(value), org[is.na(value)])) %>%
+      select(-org) ->
+      L101.inNBS_Mtce_province_S_F
+
+    # Make adjustments to  Tibet (XZ) which is mostly unrepresented in the CESY
+    # We have estimates of Tibet shares of energy use by sector.  A rough estimate to give scale
+    # Tibet accounts for 1% of the national energy.
+    Tibet_share %>%
+      left_join(tibet_shares_mappings, by = c("xz.sector", "xz.fuel")) %>%
+      filter(!grepl('biomass', fuel)) ->
+      L101.tibet_pct
+
+    L101.inNBS_Mtce_province_S_F %>%
+      group_by(year) %>%
+      filter(!is.na(value)) %>%
+      summarise(value=sum(value)) %>%
+      ungroup() ->
+      L101.inNBS_Mtce_CHINA
+
+    L101.tibet_pct %>%
+      mutate(year = 0, province = c("XZ")) %>%
+      complete(year=HISTORICAL_YEARS,nesting(province, EBProcess, EBMaterial, xz.sector, xz.fuel, sector, fuel, share)) %>%
+      left_join(L101.inNBS_Mtce_CHINA,by = c("year")) %>%
+      mutate(value = value * 0.01 * share, xz.sector = NULL, xz.fuel = NULL,share = NULL) ->
+      L101.inNBS_Mtce_tibet_S_F
+
+    L101.NBS_use_all_Mtce %>%
+      filter(province != "XZ") %>%
+      bind_rows(L101.inNBS_Mtce_tibet_S_F) %>%
+      complete(year=HISTORICAL_YEARS,nesting(province, EBProcess, EBMaterial, sector, fuel)) %>%
+      #In the test, if the first data is NA, it will be considered that the type of the entire column is a character, and a type error occurs.
+      arrange(desc(year))->
+      L101.NBS_use_all_Mtce
+
+    L101.inNBS_Mtce_tibet_S_F %>%
+      group_by(province,year,sector, fuel) %>%
+      filter(!is.na(value)) %>%
+      summarise(value = sum(value)) %>%
+      ungroup() ->
+      L101.inNBS_Mtce_tibet_S_F
+
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(province != "XZ") %>%
+      bind_rows(L101.inNBS_Mtce_tibet_S_F) %>%
+      complete(nesting(province, sector, fuel), year = c(HISTORICAL_YEARS))  ->
+      L101.inNBS_Mtce_province_S_F
+
+    # Produce outputs
+    L101.NBS_use_all_Mtce %>%
+      add_title("NBS china energy statistical yearbook by sector / fuel / year") %>%
+      add_units("Unit = Mtce") %>%
+      add_comments("historical years") %>%
+      add_legacy_name("L101.NBS_use_all_Mtce") %>%
+      add_precursors("gcam-china/province_names_mappings",
+                     "gcam-china/NBS_CESY_process",
+                     "gcam-china/NBS_CESY_material",
+                     "gcam-china/en_balance_Mtce_Yh_province",
+                     "gcam-china/Tibet_share",
+                     "gcam-china/tibet_shares_mappings") %>%
+      add_flags(FLAG_PROTECT_FLOAT, FLAG_SUM_TEST) ->
```

Comment:
I compared the old and new results and found some very small differences after the interpolation. （1e-14）

Reply:
@pralitp or @kvcalvin  would this be a problem?